src/actors/client/ActorClient/ActorClientGRPC.ts

```diff
@@ -0,0 +1,278 @@
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Any } from "google-protobuf/google/protobuf/any_pb";
+import { ExecuteActorStateTransactionRequest, GetActorStateRequest, GetActorStateResponse, GetMetadataResponse, InvokeActorRequest, InvokeActorResponse, RegisterActorReminderRequest, RegisterActorTimerRequest, TransactionalActorStateOperation, UnregisterActorReminderRequest, UnregisterActorTimerRequest } from '../../../proto/dapr/proto/runtime/v1/dapr_pb';
+import GRPCClient from '../../../implementation/Client/GRPCClient/GRPCClient';
+import { OperationType } from '../../../types/Operation.type';
+import { ActorReminderType } from '../../../types/ActorReminder.type';
+import { ActorTimerType } from '../../../types/ActorTimer.type';
+import IClientActor from '../../../interfaces/Client/IClientActor';
+import { KeyValueType } from '../../../types/KeyValue.type';
+
+// https://docs.dapr.io/reference/api/actors_api/
+export default class ActorClientGRPC implements IClientActor {
+  client: GRPCClient;
+
+  constructor(client: GRPCClient) {
+    this.client = client;
+  }
+
+  async invoke(method: "GET" | "POST" | "PUT" | "DELETE", actorType: string, actorId: string, methodName: string, body?: any): Promise<object> {
+    const msgService = new InvokeActorRequest();
+    msgService.setActorId(actorId)
+    msgService.setActorType(actorType);
+    msgService.setMethod(methodName);
+
+    if (body) {
+      // @todo: if body is any, do we have to figure out how to serialize in JS? (e.g. if object -> JSON.stringify?)
+      msgService.setData(body);
+    }
+
+    return new Promise((resolve, reject) => {
+      const client = this.client.getClient();
+      client.invokeActor(msgService, (err, res: InvokeActorResponse) => {
+        if (err) {
+          return reject(err);
+        }
+
+        // https://docs.dapr.io/reference/api/secrets_api/#response-body
+        const resData = Buffer.from(res.getData()).toString();
+
+        try {
+          return resolve(JSON.parse(resData));
+        } catch (e) {
+          return resolve(resData as any);
+        }
+      });
+    });
+  }
+
+  async stateTransaction(actorType: string, actorId: string, operations: OperationType[]): Promise<void> {
```

Comment:
In the implementation in the Java SDK, we keep a separate client for the "server" side of actor and the "client" side of actors. It basically means that the invoke API is in a separate interface and implementation. It is important to keep the abstraction between client and server.